index.bs

```diff
@@ -6145,6 +6244,354 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+
+: Client extension processing
+:: If {{AuthenticationExtensionsClientInputs/devicePubKey}} is present, the client creates the authenticator extension input from the client extension input.
+
+: Client extension output
+:: An ArrayBuffer containing the [=device public key attestation object=] that was returned in the authenticator extension output and the signature returned as the [=unsigned extension output=].
+    <xmp class="idl">
+    dictionary AuthenticationExtensionsDevicePublicKeyOutputs {
+        ArrayBuffer authenticatorOutput;
+        ArrayBuffer signature;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        AuthenticationExtensionsDevicePublicKeyOutputs devicePubKey;
+    };
+    </xmp>
+
+
+: Authenticator extension input
+:: A CBOR expression of the client extension input
+
+    ```
+    devicePublicKeyInputs = {
+      attestation: "none" / "indirect" / "direct" / "enterprise",
+      attestationFormats: [tstr],
+    }
+    $$extensionInput //= (
+      devicePubKey: devicePublicKeyInputs,
+    )
+    ```
+
+: Authenticator extension output
+:: The <dfn>device public key attestation object</dfn>, defined by the `attObjForDevicePublicKey` type:
+
+    ```
+    $$extensionOutput //= (
+        devicePubKey: attObjForDevicePublicKey,
+    )
+
+    attObjForDevicePublicKey = { ; Note: This object conveys an attested
+                                 ; device public key and is analogous to \`attObj\`.
+
+        aaguid:  bstr,  ; Authenticator's AAGUID (16 bytes fixed-length)
+                        ; https://www.w3.org/TR/webauthn/#aaguid
+
+        dpk:     bstr,  ; The Device Public Key (self-describing variable length,
+                        ; COSE_Key format, CBOR-encoded)).
+
+        ; Whether this key is scoped to the entire device, or a loosely-defined,
+        ; narrower scope called "app". For example, a "device"-scoped key is expected
+        ; to be the same between an app and a browser on the same device, while
+        ; an "app"-scoped key would probably not be.
+        ;
+        ; Whatever the scope, a device key is still specific to a given credential
+        ; and does not provide any ability to link credentials.
+        ;
+        ; Whether device-scoped or not, keys are still device-bound. I.e. an
+        ; app-scoped key does not enjoy lesser protection from extraction.
+
+        scope: uint .size 1, ; a value of 0x00 means "entire device" ("all apps") scope.
+                             ; 0x01 means "per-app" scope.
+                             ; Values other than 0x00 or 0x01 are reserved for future
+                             ; use.
+
+        ; An authenticator-generated random nonce for inclusion in the attestation
+        ; signature. If the authenticator chooses to not generate a nonce, it sets this
+        ; to a zero-length bytestring.
+
+        nonce: bstr .size (0..32),
```

Comment:
What are the security implications of having vs not-having this nonce? If there are none, why have it at all? If there is a good reason, which I expect there is, then I feel it should be outlined here.